src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -2632,6 +2800,19 @@ namespace winrt::TerminalApp::implementation
                                                   const winrt::TerminalApp::TabBase& sourceTab,
                                                   TerminalConnection::ITerminalConnection existingConnection)
     {
+        // First things first - Check for making a pane from content GUID.
+        if (newTerminalArgs &&
+            newTerminalArgs.ContentId() != 0)
+        {
+            // Don't need to worry about duplicating or anything - we'll
+            // serialize the actual profile's GUID along with the content guid.
+            const auto& profile = _settings.GetProfileForArgs(newTerminalArgs);
+
+            const auto control = _InitControlFromContent(newTerminalArgs.ContentId());
+            _RegisterTerminalEvents(control);
```

Comment:
hmm why doesn't `initcontrol` register events? it sounds like it should do anything required to initialize.